File: src/main/字符串/t389/FindTheDifference.java
package 字符串.t389;

/**
 * @author: sotowang
 * @date: 2020/12/18 15:49
 */
public class FindTheDifference {
    public char findTheDifference(String s, String t) {
        int[] arr = new int[26];
        char[] chars = s.toCharArray();
        for(int i=0;i<chars.length;i++){
            arr[chars[i]-'a']++;
        }
        char[] tchars = t.toCharArray();
        for(int i=0;i<tchars.length;i++){
            if(arr[tchars[i]-'a']==0){
                return tchars[i];
            }
            arr[tchars[i]-'a']--;
        }
        return ' ';
    }
}

File: src/main/字符串/t389/findTheDifference.go
package t389

func findTheDifference(s string, t string) byte {
	arr := make([]int, 26)
	n := len(s)
	for i := 0; i < n; i++ {
		arr[s[i]-'a']++
	}
	n = len(t)
	for i := 0; i < n; i++ {
		if arr[t[i]-'a'] == 0 {
			return t[i]
		}
		arr[t[i]-'a']--
	}
	return ' '
}
